### ISSUE TITLE
A few 'potential' issues in Rotation example

### ISSUE THREAD

**Issue by BrianCollinss** [APSIMInitiative/ApsimX#9096]
### What happened?

The legend of the yield plot looks strange:

![image](https://github.com/APSIMInitiative/ApsimX/assets/39573622/7c2168d5-3b0c-43a7-a61b-f43f631de816)

When I added a 'RotationRugplot' under 'RotationManager' (and I am not quite familiar with this type of plot), I get the following error message immediately and the plot is blank (even if sims have been run already and DB is written):

`No rule/value table in storage.`

If I try to run APSIM while 'RotationRugplot' is selected, it does not run. I have to select 'RotationManager' or another node before I can run the model. When I run, then  'RotationRugplot' displays its content.

When I remove 'RotationRugplot' and try to re-run, APSIM breaks: 

![image](https://github.com/APSIMInitiative/ApsimX/assets/39573622/6d73fa4d-e781-400e-8c46-31bc94132b0d)

### What Operating System are you on?

Windows

**Comment by par456** [comment]
@peter-devoil 

This might be one for you. I'm not overly familiar on how rugplot works, but I was able to replicate everything Brian did. Namely that once the exception is thrown, you have to click off the rugplot before you can click run and that if you remove the rugplot the currentpaddock is cleared and the simulation won't run anymore.

I also note that it doesn't seem to work well if the rotation manager is within an experiment, as you can only view the rugplot of the first simulation in the experiment. Being able to change that would be good.

If you haven't got much time I'll have a crack at solving this, but I figured you might want to try first.

**Comment by peter-devoil** [comment]
I'll have a look later, and add the simulation selection code. It's quite specific to rotations, particularly the "rules" that the rotation graph evaluates every day (sometimes several times).

**Comment by BrianCollinss** [comment]
Is there any random sampling used by 'RotationManager' for any reason? Every time I run, 'TransitionReport' changes.

![image](https://github.com/APSIMInitiative/ApsimX/assets/39573622/fee91434-1edc-45ba-9b95-2cff8e003355)

![image](https://github.com/APSIMInitiative/ApsimX/assets/39573622/6015156c-4aee-4e82-bd99-fba723fedf35)
![image](https://github.com/APSIMInitiative/ApsimX/assets/39573622/9ea7fe2c-0526-4e24-adfb-8b261c4ca116)
![image](https://github.com/APSIMInitiative/ApsimX/assets/39573622/949c4364-e774-47f9-b792-f94bbf64cc06)

**Comment by peter-devoil** [comment]
Not unless your manager rules are stochastic. As it doesnt have a way to switch between simulations, it's may be garbage in a multi-simulation configuration.

**Comment by BrianCollinss** [comment]
No I'm using the original example file.

**Comment by peter-devoil** [comment]
Looking at [this bit](https://github.com/APSIMInitiative/ApsimX/blob/9b60bbc04e629a7cf37f53f715a5e149d7781927/Models/Management/RotationLogger.cs#L289) makes me think the similation it presents will be randomly chosen from whatever simulations are there

**Comment by BrianCollinss** [comment]
@peter-devoil I didn't pay attention to the fact that this is an Experiment! Sorry.

**Comment by peter-devoil** [comment]
@par456 [this](https://github.com/APSIMInitiative/ApsimX/blob/9b60bbc04e629a7cf37f53f715a5e149d7781927/Models/Management/RotationLogger.cs#L371) was unsuccessfully stolen from somewhere else (summary file, most likely) - is there a better way to find the simulation names within a simulations object?

**Comment by par456** [comment]
Not really, I don't think we have a utilities tool for doing that, and that code will do the job you need.
